transformations/yoda_transform/transformation.py

```diff
@@ -0,0 +1,130 @@
+import itertools
+
+import spacy
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Base Class for implementing the different input transformations a generation should be robust against.
+"""
+"""
+Yoda Speak implementation borrowed from here: https://github.com/yevbar/Yoda-Script, with minor mofifications to fix
+deprecation and to change some stylistic features
+"""
+
+punctuation = [",", ".", ";", "?", "!"]
+nlp = spacy.load("en_core_web_sm")
+comma = nlp(" , ")[1]
+
+
+def sentify(text):
+    output = []
+    doc = nlp(text)
+    for sent in doc.sents:
+        sentence = []
+        for clause in clausify(sent):
+            sentence.append(yodafy(clause))
+        output.append(sentence)
+    return output
+
+
+def clausify(sent):
+    output = []
+    cur = []
+    for token in sent:
+        if token.dep_ == "cc" or (
+            token.dep_ == "punct" and token.text in punctuation
+        ):
+            output.append(cur)
+            output.append([token])
+            cur = []
+        else:
+            cur.append(token)
+    if cur != []:
+        output.append(cur)
+    return output
+
+
+def yodafy(clause):
+    new_array = []
+    state = False
+    for token in clause:
+        if state:
+            new_array.append(token)
+        if not state and (token.dep_ == "ROOT" or token.dep_ == "aux"):
+            state = True
+    if len(new_array) > 0 and new_array[len(new_array) - 1].dep_ != "punct":
+        new_array.append(comma)
+    for token in clause:
+        new_array.append(token)
+        if token.dep_ == "ROOT" or token.dep_ == "aux":
+            break
+    return new_array
+
+
+def yoda(s):
+    string = []
```

Comment:
Please add [PEP 257](https://www.python.org/dev/peps/pep-0257/) style comments to all the functions.